### PR TITLE
Follow Python 3.9 on AT next

### PR DESCRIPTION
**Pull request by mscastanho**
This PR includes a fix for Python FVTR tests on AT next, an update to Python 3.9 on AT next and a few changes in the way we distribute Python. Copying the commit message of the most relevant commit:
    
    Python 3.9 has been recently released and is the version that should be
    followed by AT 15 (current AT next). While updating the revision, I took the
    time to fully revise all patches we have been applying to Python and actually
    found out we don't need two of them anymore (with a catch).
    
    1) Patch to skip test_ssl.test_random_fork:
    
        This was needed back in 2017 due to a bug in OpenSSL 1.1.1c (see discussion
        and linked Python issues in [0]). This has been fixed upstream a long time
        ago, and AT next currently follows OpenSSL 1.1.1h, which already includes
        the fix.
    
    2) Patch to update Python's lib path to AT's:
    
        Previous AT versions used to distribute all Python modules and packages
        under a python* directory on /opt/at<XX>/lib64. That is not default behavior
        upstream, so we needed this patch to force AT's Python to look for stuff in
        that subdir instead of the default ones under lib/.
    
        There are differences in the way distros approach this. The default scheme
        of installing everything to /usr/lib/ is used by Debian and Ubuntu. SLES and
        Fedora, on the other hand, split Python stuff into lib and lib64. All
        generic and pure-Python code is installed on lib and any platform-specific
        code goes to lib64. That behavior was not supported by Python upstream's
        install scripts, and these distros had to maintain off-tree patches (like
        ours) for a long time [1]. This changed in Python 3.9, which added the
        --with-platlibdir flag to help Fedora and SLES properly install stuff the
        way they need without needing off-tree patches.
    
        The split into the two different directories is specially useful for
        situations when one needs to install Python built for different targets on
        the same system. This way a 32- and a 64-bit installation can share the
        generic files and only install the platform-specific stuff in their
        corresponding directories. That obviously is not the case for AT, as we
        don't support 32-bit targets anymore.
    
        Still, I'd prefer if AT used an installation method that is supported
        upstream, so we can reduce the amount of off-tree patches we need to
        maintain, specially a big and complex one like this. They usually need
        updates now and then when a new version makes incompatible changes.
        Splitting the installation into lib and lib64 dirs as described above makes
        sense and should work well for AT.
    
        So instead of applying our off-tree patch, this commit configures Python
        with --with-platlibdir=lib64, which as a side-effect changes the way AT
        distributes Python modules and packages:
          Before: everything under /opt/at<XX>/lib64/
          After: generic and pure-Python modules /opt/at<XX>/lib and platform
          specific files in /opt/at<XX>/lib64.
    
        This change should be transparent to AT users as Python configured this way
        knows where it should look for stuff.
    
    [0] https://github.com/advancetoolchain/advance-toolchain/issues/201
    [1] https://bugs.python.org/issue1294959

Example of how Python stuff is distributed in AT 14:
``` bash
$ find /opt/at14.0/ -name '*python*' -type d
/opt/at14.0/lib/debug/opt/at14.0/lib64/python3.8
/opt/at14.0/lib64/python3.8
/opt/at14.0/lib64/cmake/boost_python-1.73.0
/opt/at14.0/share/gcc-10.2.1/python
/opt/at14.0/share/gdb/python
/opt/at14.0/include/python3.8
/opt/at14.0/include/python3.8/cpython
/opt/at14.0/include/boost/mpi/python
/opt/at14.0/include/boost/parameter/aux_/python
/opt/at14.0/include/boost/python
```

Now on AT next after this change:
```
$ find ./opt/at-next-15.0-0-alpha/ -name '*python*' -type d
./opt/at-next-15.0-0-alpha/lib64/cmake/boost_python-1.73.0
./opt/at-next-15.0-0-alpha/lib64/python3.9
./opt/at-next-15.0-0-alpha/include/python3.9
./opt/at-next-15.0-0-alpha/include/python3.9/cpython
./opt/at-next-15.0-0-alpha/include/boost/mpi/python
./opt/at-next-15.0-0-alpha/include/boost/parameter/aux_/python
./opt/at-next-15.0-0-alpha/include/boost/python
./opt/at-next-15.0-0-alpha/share/gcc-11.0.0/python
./opt/at-next-15.0-0-alpha/share/gdb/python
./opt/at-next-15.0-0-alpha/lib/python3.9
```

Closes #1763 
Closes #1818

### CODE REVIEW

**Comment by tuliom** [comment]
I miss the directory with the debug information:
```
/opt/at14.0/lib/debug/opt/at14.0/lib64/python3.8
```
Did it really disappear or is just just the difference between installing via RPM and building manually?

**Comment by mscastanho** [comment]
@tuliom I think the debug dir is a symlink, and it's only created during the install step.
```
$ find ./opt/at-next-15.0-0-alpha/ -name 'debug' -type d
./opt/at-next-15.0-0-alpha/lib64/go/11.0.0/powerpc64le-linux-gnu/debug
./opt/at-next-15.0-0-alpha/include/c++/11.0.0/debug
./opt/at-next-15.0-0-alpha/include/boost/bimap/detail/debug
./opt/at-next-15.0-0-alpha/include/boost/preprocessor/debug
./opt/at-next-15.0-0-alpha/include/boost/spirit/home/classic/debug
```

**Comment by mscastanho** [comment]
Update to also apply the FVTR test fix to gdb, which suffered from the same issue.

**Comment by tuliom** [comment]
> Did it really disappear or is just just the difference between installing via RPM and building manually?

Confirmed! This is just a different between comparing an installed AT vs. a directory created at build time.
So, the debug directory was not affected by this change.
Files are under `opt/at-next-15.0-0-alpha/lib64/python3.9`.

**Review by tuliom — APPROVED** [review]
LGTM.
Let's just wait for all the tests to pass before merging it.

File: fvtr/gdb/gdb.exp
Comment (ThinkOpenly) [inline fvtr/gdb/gdb.exp:30]: Completely subjective and perhaps personal preference, but would `^` work here instead of `\A`? The caret (`^`) is more familiar to me, at least.
(Same for the next change, as well.)

**Review by ThinkOpenly — APPROVED** [review]
For the commit "Fix Python and gdb FVTR tests on AT next", might I suggest a new subject "Fix to not exclude Python and gdb FVTR tests", since you are not fixing the tests themselves. Not a big deal, though.

File: fvtr/gdb/gdb.exp
Comment (mscastanho) [reply fvtr/gdb/gdb.exp:30]: @ThinkOpenly It does, but apparently requires an extra switch (`-lineanchor`). But I agree that is more standard and can make it less confusing to future us/others taking a look at this code.

But since tests are already running (and they take a long time), I'll leave that to a future PR addressing #1813.

**Comment by mscastanho** [comment]
The `test_subprocess` Python test failed on RHEL 8. I'm taking a closer look to understand what happened.

**Comment by tuliom** [comment]
> The `test_subprocess` Python test failed on RHEL 8. I'm taking a closer look to understand what happened.

@mscastanho Reminder: that's the test that used to timeout.

**Comment by mscastanho** [comment]
@tuliom Yes, I'm aware. It timed out again, but on a different part. I'm increasing the timeout and testing it again.

**Comment by mscastanho** [comment]
I updated the timeout for another Python test, rebased the PR against master and also updated the commit title as suggested by @ThinkOpenly.

**Comment by mscastanho** [comment]
Info about failed tests:
- AT 14.0 RHEL8 ppc64le: server under high load and some processes got killed. Trying again.
- AT * SLES12 ppc64le: server was misconfigured and `oprofile` test failed on all of them. Fixed and trying again.
- AT 11 Ubuntu16 ppc64le: intermittent Python test failures.
- AT 12 Debian9 ppc64le: same...

**Comment by mscastanho** [comment]
Last failing tests:
- AT 12 Debian9: Rebuilding to see if we avoid the intermittent failures this time
- AT 10 SLES12: One python test is consistently failing: `test_strptime`. This test has been fixed on Python 3.6 (AT >= 11) by python/cpython@f1b9ad3d38c11. That specific build server has been update recently and probably altered some timezone files that triggered the issue. I'll include a commit to skip this test on AT 10 on an upcoming PR.

**Comment by mscastanho** [comment]
AT 12 Debian9 build passed.

**Pull request merged**
